### PR TITLE
Bugfix/missing attributes for dynamic proxy

### PR DESCRIPTION
**Pull request by huoshan12345**
1. replicate attributes **that are targeted on implementation type** to proxy type in method `ProxyGeneratorUtils.CreateClassProxyInternal`, which is used in method `ProxyGenerator.CreateClassProxy<T>()`
2. add tests for 2#
3. fix typo `CustomAttributeBuildeUtils` -> `CustomAttributeBuilderUtils`
4. `typeof(IAspectValidator).Name` -> `nameof(IAspectValidator)`

### CODE REVIEW

**Comment by huoshan12345** [comment]
**Here is a related question:** 
Do we need to replicate attributes(targeted on type, property, method) from implementation type to proxy type that are created from `ProxyGenerator.CreateClassProxy<TService, TImplementation>`?
Currently, 
1. all the attributes from `TService` are replicated to proxy type.
2. none of the attributes from `TImplementation` is replicated to proxy type.

Personally, I think we can follow the same logic as `TService service = new TImplementation()`.
In other words,
1. **donot** replicate any attributes from `TService` to proxy type, which should be inherited to proxy type.
2. replicate all the attributes from `TImplementation` to proxy type.

**Review by liuhaoyang — APPROVED** [review]
